lib/web_ui/lib/src/engine/text_editing/input_type.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// Various types of inputs used in text fields.
+///
+/// These types are coming from Flutter's [TextInputType]. Currently, we don't
+/// support all the types. We fallback to [EngineInputType.text] when Flutter
+/// sends a type that isn't supported.
+// TODO(flutter_web): Support more types.
+abstract class EngineInputType {
+  const EngineInputType();
+
+  static EngineInputType fromName(String name) {
+    switch (name) {
+      case 'TextInputType.number':
+        return number;
+      case 'TextInputType.phone':
+        return phone;
+      case 'TextInputType.emailAddress':
+        return emailAddress;
+      case 'TextInputType.url':
+        return url;
+      case 'TextInputType.multiline':
+        return multiline;
+
+      case 'TextInputType.text':
+      default:
+        return text;
+    }
```

Comment:
There is also a datetime type. https://api.flutter.dev/flutter/services/TextInputType-class.html

Is there a reason for omitting it? Shall we add a comment or to do?

Reply:
The only reason is that it's not supported on the web (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode). Anyway, I doubt people use this input type often. They probably use some kind of a date picker (e.g. `material/date_picker.dart`).